packages/medusa/src/interfaces/abstract-parser.ts

```diff
@@ -0,0 +1,42 @@
+import { ParseConfig } from "papaparse"
```

Comment:
comment: In fact, I don't think that should be part of that file. It is really tightly link to csv parsing and we are at the abstract level here. maybe here, the ParserOptions should just be template?

Reply:
I thought of 3 options to solve this:
1. Define a type ourselves which can serve as an interface. Different implementations of the `AbstractParser` class will  have to adhere to the `ParseOptions` interface that we have defined ourselves. This means that we'll probably have to map some of the options to whatever underlying csv library the class (i.e: `CsvParser`) will be using.
2. Define an extra generic in the `IParser<TResult, TParseOptions>` interface and make use of it at the moment the implementation of the `AbstractParser` is written. Something like this:
```typescript
import Papa, { ParseConfig } from "papaparse"

abstract class AbstractParser<
  TSchema,
  TParserResult,
  TParseOptions, // new generic
  TOutputResult
> implements IParser<TParserResult, TParseOptions> {
  // ... 
  public abstract parse(
    readableStream: NodeJS.ReadableStream,
    options?: TParseOptions
  ): Promise<TParserResult[]>
}

class CsvParser<
  TSchema extends CsvSchema = CsvSchema,
  TParserResult = unknown,
  TOutputResult = unknown
> extends AbstractParser<TSchema, TParserResult, ParseConfig, TOutputResult> {
  // ...
  
  public async parse(
    readableStream: NodeJS.ReadableStream,
    options: ParseConfig = DEFAULT_PARSE_OPTIONS
  ): Promise<TParserResult[]> {}
}
```
3. Use `any` or `unknown` for the options

Let me know what you think 😄 !

Reply:
👌 i would go with the second option for now wdyt?

Reply:
makes sense! done :smile: